oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/web/server/OAuth2AuthorizationRequestRedirectWebFilterTests.java

```diff
@@ -53,6 +60,15 @@
 	@Mock
 	private ServerAuthorizationRequestRepository<OAuth2AuthorizationRequest> authzRequestRepository;
 
+	@Mock
+	private ServerRequestCache requestCache;
+
+	@Spy
+	private WebSessionServerRequestCache spiedRequestCache;
```

Comment:
Let's change this to a `ServerRequestCache` and mock out the methods. This ensures the test isn't relying on `WebSessionServerRequestCache` behavior

Reply:
Sorry for the comings and goings here @rwinch , but I'd like to make sure we're on the same page.
Wouldn't that be what I first did on the [filterWhenExceptionThenRedirected](https://github.com/spring-projects/spring-security/pull/6418/files#diff-6eb7b3bb0614ba3228ea67da35a04b9aL132) test?
I first chose to modify that test instead of creating a new one, since that will be the actual scenario that we want to test, i.e. that when we receive a `ClientAuthorizationRequiredException`, we redirect and save the original request in the process.
I can create a new test hat verifies that the `ServerRequestCache` is used, but it will be pretty much the same test that the one I mentioned above, wouldn't it? Please let me know if I'm missing something here :). I'll clean everything once I understand your perspective correctly, sounds good?

Reply:
Yes it is quite similar. However, please use a Mock instead of a Spy. With the Spy we are relying on behavior from `WebSessionServerRequestCache` to see if things are working. For example, the assertion is checking to see if a session attribute has been set or not. This means that this test requires `WebSessionServerRequestCache` sets that attribute. All we do care is that the save method is invoked with the correct arguments.

Reply:
Of course, I agree that we don't want an integration test here.
Let me try to explain better, I now removed the integration test and added a *new* test. I believe that's the test that you had in mind, correct? 
My question is, does this test bring any value to the table? It's exactly the same scenario that for the [filterWhenExceptionThenRedirected](https://github.com/spring-projects/spring-security/pull/6418/files#diff-6eb7b3bb0614ba3228ea67da35a04b9aL132) isn't it?
That's why I considered that we don't need a new test, but just adding the validation to the pre existing test would be enough, but please correct me if I'm wrong, I'm probably missing something here :)